### DISCUSSION TITLE
Locked Property Prevents Changing Not Locked Property

### DISCUSSION THREAD

**Discussion by stebers** [livewire/livewire#6917] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

If there is a locked property and other not locked properties that have the name of the locked property as prefix, the locked property will lock those other properties as well.

1. Create a livewire component with 2 properties
```php
#[Locked]
public string $filter = '';

public string $filterRule = '';
```  

2. Create an input field with attribute `wire:model.live="filterRule"`
3. Type some value in this input field
4. Notice error: **Cannot update locked property: [filter]**

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment, marked as answer]
Thanks for reporting. This is indeed a bug. I've added a failing test:
https://github.com/livewire/livewire/pull/6925

**Reply by lrljoe** [reply]
(Not Fully Tested - by me at least) Fix PR
[https://github.com/livewire/livewire/pull/6926](https://github.com/livewire/livewire/pull/6926)
